e2e/testcases/helm_sync_test.go

```diff
@@ -187,6 +187,90 @@ func TestHelmDefaultNamespace(t *testing.T) {
 	}
 }
 
+// TestHelmLatestVersion verifies the Config Sync behavior for helm charts when helm.spec.version is not specified. The helm-sync
+// binary should pull down the latest available version in this case. It also tests that if a new helm chart gets pushed, the
+// chart version gets automatically updated by Config Sync.
+// This test will work only with following pre-requisites:
+// Google service account `e2e-test-ar-reader@${GCP_PROJECT}.iam.gserviceaccount.com` is created with `roles/artifactregistry.reader` for accessing images in Artifact Registry.
+// A JSON key file is generated for this service account and stored in Secret Manager
+func TestHelmLatestVersion(t *testing.T) {
+	nt := nomostest.New(t,
+		nomostesting.SyncSource,
+		ntopts.Unstructured,
+		ntopts.RequireGKE(t),
+	)
+
+	rs := fake.RootSyncObjectV1Beta1(configsync.RootSyncName)
+	nt.T.Log("Fetch password from Secret Manager")
+	key, err := gitproviders.FetchCloudSecret("config-sync-ci-ar-key")
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Create secret for authentication")
+	_, err = nt.Shell.Kubectl("create", "secret", "generic", "foo", fmt.Sprintf("--namespace=%s", v1.NSConfigManagementSystem), "--from-literal=username=_json_key", fmt.Sprintf("--from-literal=password=%s", key))
+	if err != nil {
+		nt.T.Fatalf("failed to create secret, err: %v", err)
+	}
+	nt.T.Cleanup(func() {
+		nt.MustKubectl("delete", "secret", "foo", "-n", v1.NSConfigManagementSystem, "--ignore-not-found")
+	})
+
+	remoteHelmChart, err := helm.PushHelmChart(nt, privateSimpleHelmChart, privateSimpleHelmChartVersion)
+	if err != nil {
+		nt.T.Fatalf("failed to push helm chart: %v", err)
+	}
+
+	nt.T.Log("Update RootSync to sync from a private Artifact Registry")
+	nt.MustMergePatch(rs, fmt.Sprintf(`{"spec": {"sourceType": "%s", "git": null, "helm": {"repo": "%s", "chart": "%s", "auth": "token", "version": "", "namespace": "", "deployNamespace": "simple", "secretRef": {"name" : "foo"}}}}`,
+		v1beta1.HelmSource, helm.PrivateARHelmRegistry, remoteHelmChart.ChartName))
+	err = nt.WatchForAllSyncs(nomostest.WithRootSha1Func(helmChartVersion(remoteHelmChart.ChartName+":"+privateSimpleHelmChartVersion)),
```

Comment:
The assertions with `WatchForAllSyncs`/`Validate` here and below could be simplified to the following (with the updated version in the latter assertion):
```
nt.Watcher.WatchObject(
  kinds.Deployment(), "deploy-default", "simple",
  []testpredicates.Predicate{testpredicates.HasLabel("version", privateSimpleHelmChartVersion)}
)
```
This will wait for the deployment to have the label in the predicate

Reply:
Thanks for the pointer! The test looks a lot simpler now.

Reply:
No problem! Agreed looks good